implementations/micrometer-registry-otlp/src/main/java/io/micrometer/registry/otlp/OtlpStepTuple2.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright 2023 VMware, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.micrometer.registry.otlp;
+
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.step.StepTuple2;
+
+import java.util.function.Supplier;
+
+/**
+ * This is an internal method not meant for general use. The only reason to have this
+ * class so that {@code OtlpMeterRegistry} can call {@code _closingRollover} on
+ * {@code StepTuple2} and the method does not need to be public.
+ */
+class OtlpStepTuple2<T1, T2> extends StepTuple2<T1, T2> {
```

Comment:
Now that StepTuple2 has been copied over to OTLP. I guess the better this would be for this to support StepN, which can hold count, total, max, (maybe step-histogram too) so that it makes sure that complete rotation of StepMeter happens with minimal overhead. 
Do you think will it be hard to have StepHolder which can hold 'N' items aligned to same step and should be rotated together?

Reply:
The concern I have is instead of `poll1` and `poll2` there should be a `poll(index)` as well as you should provide a collection of "noValue" and a collection of `Supplier`.
I think this can be chaotic rather quickly.

I would like to look into improving step-histogram and max (we don't have an interface for that), so that we can make things simpler and more robust. Once we are there, this idea can be handy but right now I think I would prefer separate types.